src/services/UserService.ts

```diff
@@ -21,4 +21,31 @@ export default class UserService {
     users.push(newUser);
     return newUser;
   }
+
+  async updateUser(id: string, user: User): Promise<User> {
+    const { username } = user;
+    const updatedUser = await usersData.find(obj => obj.id === id);
+    if (!updatedUser) {
+      throw new HttpException('User not found', HttpStatus.BAD_REQUEST);
+    }
+    // save data
+    usersData = usersData.map(obj => {
+      if (obj.id === id) {
+        obj = { ...obj, username }
```

Comment:
It's okay but think about the user with lots of info (first name, last name, address, cell number, date of birth, etc) :)